component/identity/src/main/java/org/exoplatform/services/organization/idm/UserDAOImpl.java

```diff
@@ -915,4 +931,113 @@ private boolean filterDisabledUsersInQueries() {
   private boolean disableUserActived() {
     return orgService.getConfiguration().isDisableUserActived();
   }
+
+  private void saveLastLoginFail(String username) {
+    try {
+      User user = orgService.getUserHandler().findUserByName(username);
+      if (user != null) {
+        UserProfile profile = orgService.getUserProfileHandler().findUserProfileByName(username);
+        if (profile == null) {
+          profile = orgService.getUserProfileHandler().createUserProfileInstance(username);
+        }
+        int currentNbFail = profile.getAttribute(BAD_PASSWORD_COUNT) != null ?
+                            Integer.parseInt(profile.getAttribute(BAD_PASSWORD_COUNT)) : 0;
+        currentNbFail++;
+        profile.setAttribute(BAD_PASSWORD_COUNT, String.valueOf(currentNbFail));
+        Instant now = Instant.now();
+        profile.setAttribute(BAD_PASSWORD_TIME, String.valueOf(now.toEpochMilli()));
+        orgService.getUserProfileHandler().saveUserProfile(profile, true);
+
+        if (currentNbFail >= orgService.getConfiguration().getMaxBadPasswordCount()) {
```

Comment:
how about sending an analytics event also?

Reply:
analytics added for failed login, and locked account